torch/csrc/autograd/python_variable.cpp

```diff
@@ -914,6 +914,12 @@ int THPVariable_set_grad(THPVariable* self, PyObject* py_grad, void* unused) {
       "can't assign Variable as its own grad");
 
   const auto& grad = THPVariable_Unpack(py_grad);
+  TORCH_CHECK_TYPE(
+      var.dtype() == grad.dtype(),
+      "assigned grad has data of type ",
+      grad.dtype(),
+      " that differs from the required type ",
```

Comment:
made a small modification: "attempting to assign a gradient with dtype A to a tensor with dtype B"

Reply:
actually maybe also something like "please ensure that the grad and tensor have the same dtype" after